program/rust/src/log.rs

```diff
@@ -15,8 +14,7 @@ use solana_program::sysvar::Sysvar;
 pub fn pre_log(accounts: &[AccountInfo], instruction_data: &[u8]) -> ProgramResult {
     msg!("Pyth oracle contract");
 
-    let instruction_header: cmd_hdr =
-        deserialize_single_field_from_buffer::<cmd_hdr>(instruction_data, None)?;
+    let instruction_header: cmd_hdr = *load::<cmd_hdr>(instruction_data)?;
```

Comment:
Can nitpick you change these to 
`let instruction_header: &cmd_hdr = load::<cmd_hdr>(instruction_data)?;` (no dereference).
No other lines should need to change.